scoring/models.py

```diff
@@ -157,6 +157,75 @@ def get_questions(self) -> list[Question]:
             Post.CurationStatus.REJECTED,
         ]
 
+        if self.project and self.project.type == Project.ProjectTypes.SITE_MAIN:
+            # global leaderboard
+            if self.start_time is None or self.end_time is None:
+                raise ValueError("Global leaderboards must have start and end times")
+            questions = Question.objects.filter(
+                Q(related_posts__post__projects__add_posts_to_main_feed=True)
+                | Q(related_posts__post__default_project__add_posts_to_main_feed=True)
+            )
```

Comment:
I made a small performance testing and this takes `5s` on my local machine.
Though, this one takes `0.8s` and _does not require_ `.distinct()`:
```python
Question.objects.filter(
    related_posts__post__in=Post.objects.filter_for_main_feed()
)
```

Reply:
Wow, 5 seconds is not good. Were you testing the endpoint response time, or just the `get_questions` method?
But either way, down to 0.8s is a crazy improvement.
The thing is, I was getting the whole `get_questions` method to run in `0.3s` locally when it was `2-3s` before the change.
We can talk through the details when we sync today.